### PR TITLE
Add support for UDS

### LINKED ISSUE #67: Support UDS

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
```
(.venv) ~/d/dd-apm-test-agent ❮❮❮ ddapm-test-agent --uds-path=./apm.socket
======== Running on http://0.0.0.0:8126, http://unix:/Users/kverhoog/dev/dd-apm-test-agent/apm.socket: ========
(Press CTRL+C to quit)
```

Resolves #67

### CODE REVIEW

File: ddapm_test_agent/agent.py [hunk 1]
```diff
@@ -591,6 +591,11 @@ def main(args: Optional[List[str]] = None) -> None:
             "to the agent."
         ),
     )
+    parser.add_argument(
+        "--uds-path",
+        type=str,
+        default=os.environ.get("DD_AGENT_URL") if os.environ.get("DD_AGENT_URL", "").startswith("unix://") else "/var/run/datadog/apm.socket"
```
Comment (Kyle-Verhoog) [inline ddapm_test_agent/agent.py:597]: Need to parse out path from `unix://`

Comment (Kyle-Verhoog) [inline ddapm_test_agent/agent.py:597]: Figure out if to hard fail if default socket not available (most likely will by default)

File: ddapm_test_agent/agent.py [hunk 2]
```diff
@@ -591,6 +591,11 @@ def main(args: Optional[List[str]] = None) -> None:
             "to the agent."
         ),
     )
+    parser.add_argument(
+        "--uds-path",
```
Comment (Kyle-Verhoog) [inline ddapm_test_agent/agent.py:595]: Needs documentation in the readme

File: releasenotes/notes/uds-64b11960931d8b77.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+---
+features:
+  - |
+    Add unix domain socket support.
```
Comment (Kyle-Verhoog) [inline releasenotes/notes/uds-64b11960931d8b77.yaml:4]: might be worth calling out how it's enabled (the env var at least) so the release note is more actionable.

File: README.md [hunk 4]
```diff
@@ -5,11 +5,9 @@
 
 <img align="right" src="https://user-images.githubusercontent.com/6321485/136316621-b4af42b6-4d1f-4482-a45b-bdee47e94bb8.jpeg" alt="bits agent" width="200px"/>
 
-The APM test agent is an application which emulates the APM endpoints of
-the [Datadog agent](https://github.com/DataDog/datadog-agent/) which can be used for testing Datadog APM client
-libraries.
+The APM test agent is an application which emulates the APM endpoints of the [Datadog agent](https://github.com/DataDog/datadog-agent/) which can be used for testing Datadog APM client libraries.
 
-See the [features](#Features) section for the complete list of functionalities provided.
```
[the remark this answers is not in the document]
Comment (colin-higgins) [reply README.md:12]: Just looked it up, I trusted the markdown pad dictionary too much.

File: README.md [hunk 5]
```diff
@@ -18,32 +16,42 @@ See the [Development](#Development) section for how to get the test agent runnin
 
 ## Installation
 
-The test agent can be installed from PyPI:
-
-    pip install ddapm-test-agent
-
+Once the agent is installed, you can start it like this:
```
[the remark this answers is not in the document]
Comment (colin-higgins) [reply README.md:19]: Looking at it again, it matches with the docker run idea as well, reverting.

**Review by Kyle-Verhoog — COMMENTED** [review]
Just one nit otherwise lgtm!

File: README.md [hunk 6]
```diff
@@ -19,31 +17,36 @@ See the [Development](#Development) section for how to get the test agent runnin
...
     docker run --rm\
             -p 8126:8126\
             -e CI_MODE=0\
             -v $PWD/tests/snapshots:/snapshots\
             ghcr.io/datadog/dd-apm-test-agent/ddapm-test-agent:latest
+```
 
 or from source:
-
+```
```
Comment (Kyle-Verhoog) [inline README.md:35]:
```suggestion
```

File: README.md [hunk 7]
```diff
@@ -19,31 +17,36 @@ See the [Development](#Development) section for how to get the test agent runnin
...
     pip install git+https://github.com/Datadog/dd-apm-test-agent
+```
 
+or a specific branch:
+```
```
Comment (Kyle-Verhoog) [inline README.md:40]:
```suggestion
```

File: README.md [hunk 8]
```diff
@@ -19,31 +17,36 @@ See the [Development](#Development) section for how to get the test agent runnin
...
+	pip install git+https://github.com/Datadog/dd-apm-test-agent@{branch}
+```
```
Comment (Kyle-Verhoog) [inline README.md:42]:
```suggestion
```

**Pull request merged**